scripts/refcodes_with_properties/EntryPropertyCalculator.py

```diff
@@ -0,0 +1,582 @@
+        return "organic entries or not"
+
+    def value(self,entry):
+        return entry.is_organic
+
+register(OrganicFilter)
+
+class PolymericFilter(_ComparativeFilter):
+    def __init__(self, args):
+        super(self.__class__,self).__init__(args)
+
+    @staticmethod
+    def name():
+        return "polymeric"
+
+    @staticmethod
+    def helptext():
+        return "polymeric entries or not"
+
+    def value(self,entry):
+        return entry.is_polymeric
+
+register(PolymericFilter)
+
+class AllHaveSitesFilter(_ComparativeFilter):
+    def __init__(self, args):
+        super(self.__class__,self).__init__(args)
+
+    @staticmethod
+    def name():
+        return "all atoms have sites"
+
+    @staticmethod
+    def helptext():
+        return "whether all atoms have to have sites"
+
+    def value(self,entry):
+        try:
+            return entry.crystal.molecule.all_atoms_have_sites
+        except:
+            return False
+
+register(AllHaveSitesFilter)
+
+class DisorderedFilter(_ComparativeFilter):
+    def __init__(self, args):
+
+        super(self.__class__,self).__init__(args)
+
+    @staticmethod
+    def name():
+        return "disordered"
+
+    @staticmethod
+    def helptext():
+        return "disordered entries or not"
+
+    def value(self,entry):
+        return entry.has_disorder
+
+register(DisorderedFilter)
+
+
+class AtomicWeightFilter(_RangeFilter):
+    def __init__(self,args):
+        super(self.__class__,self).__init__(args)
+
+    @staticmethod
+    def name():
+        return "atomic weight"
+
+    @staticmethod
+    def helptext():
+        return "specify a range of atomic weight (for the whole structure - not individual molecules)"
+
+    def value(self,entry):
+        try:
+            molecule = entry.crystal.molecule
+            return molecule.molecular_weight
+        except TypeError:
+            return 0.0
+
+register(AtomicWeightFilter)
+
+class AtomCountFilter(_RangeFilter):
+    def __init__(self,args):
+        super(self.__class__,self).__init__(args)
+
+    @staticmethod
+    def name():
+        return "atom count"
+
+    @staticmethod
+    def helptext():
+        return "specify a range of atom counts (for the whole structure - not individual molecules)"
+
+    def value(self,entry):
+        try:
+            molecule = entry.crystal.molecule
+            return len(molecule.atoms)
+        except TypeError:
+            return 0
+
+register(AtomCountFilter)
+
+class RotatableBondFilter(_RangeFilter):
+    def __init__(self,args):
+        super(self.__class__,self).__init__(args)
+
+    @staticmethod
+    def name():
+        return "rotatable bond count"
+
+    @staticmethod
+    def helptext():
+        return "specify the number of rotatable bonds (for the whole structure - not individual molecules)"
+
+    def value(self,entry):
+        try:
+            molecule = entry.crystal.molecule
+            return sum( x.is_rotatable for x in molecule.bonds )
```

Comment:
What if there are no bonds? Will the sum return 0 or None ?

Reply:
0

e.g.

`>>> x = [1,2,3,4]
>>> sum(y for y in x if y == 5)
0`